src/oom/oomd-util.h

```diff
@@ -61,10 +61,13 @@ bool oomd_memory_reclaim(Hashmap *h);
 /* Returns true if the amount of swap free is below the percentage of swap specified by `threshold_percent`. */
 bool oomd_swap_free_below(const OomdSystemContext *ctx, uint64_t threshold_percent);
 
-static inline int compare_pgscan(OomdCGroupContext * const *c1, OomdCGroupContext * const *c2) {
+static inline int compare_pgscan_and_memory_usage(OomdCGroupContext * const *c1, OomdCGroupContext * const *c2) {
         assert(c1);
         assert(c2);
 
+        if ((*c2)->pgscan == (*c1)->pgscan)
+                return CMP((*c2)->current_memory_usage, (*c1)->current_memory_usage);
+
         return CMP((*c2)->pgscan, (*c1)->pgscan);
```

Comment:
doesn't matter, but it looks weird comparing pgscan twice here. maybe instead do:

```c
r = CMP(…->pgscan, …->pgscan);
if (r != 0)
        return r;

return CMP(…, …);
```

that also has the nice benefit that the most relevant CMP is listed first, how it should be